### ISSUE TITLE
Upgrade react-table 

### ISSUE THREAD

**Issue by rosibaj** [kids-first/kf-portal-ui#1200]
Followup from #1197

- latest version had a breaking change so we pinned to last run without bug (`6.8.6`)
- should review the latest versions breaking change and actually upgrade the dependency
- involves both arranger UI components and the portal.
- The latest version  is `9.0.0` which from the number suggests more breaking changes than just the check boxeds 

Some helpful info on the upgrade: https://github.com/tannerlinsley/react-table/issues/1281#issuecomment-469175092

**Comment by jberube** [comment]
See https://github.com/kids-first/kf-portal-ui/pull/1449 for something that needs to be done as part of this task.

**Comment by aalex** [comment]
If we do #1540 it should fix this one as well.

**Comment by jberube** [comment]
@aalex Both can (and should) be done separately, and neither are blocking the other.

We use an old-ish version of `react-table` to avoid a regression introduced in a more recent build. The most recent version of it was not stable at the moment, so we resorted to freeze the version to `6.8.6`, work-around the bug(s) that the update would have fixed, and wait for a stable version, probably `7.0.0`.

The regression was the width of the cells resizing themselves wrongly when trying to resize a column.

**Comment by jberube** [comment]
See: https://github.com/tannerlinsley/react-table/issues/1281#issuecomment-486310231

**Comment by jberube** [comment]
TLDR;
Version 6.10.0 might solve the issues we had without need for a workaround nor freezing the version.

**Comment by jberube** [comment]
Added blocked by #1604 since both `kf-portal-ui` and `@arranger/arranger-ui` are dependant of `react-table`.

**Comment by denis-crsj** [comment]
Closing issue. Details transferred into Google Sheet : https://docs.google.com/spreadsheets/d/1U1JQZEBEMTP_07PngrmNh8rzbZICTX87Fe_2fEetAbI/edit#gid=0
